apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -176,6 +185,21 @@ static void *pthread_exit_thread(void *param)
 	return NULL;
 }
 
+/**
+* @fn                   :self_pthread_join_n_exit
+* @brief                :utility function for tc_pthread_pthread_create_exit_join
+* @return               :void *
+*/
+static void *self_pthread_join_n_exit(void *param)
+{
+	pid_t pid = getpid();
+
+	chk_val = pthread_join((pthread_t)pid, NULL);
```

Comment:
do we need to check return value of chk_val

Reply:
This case is checking "attempt to join to itself (deadlock)". 
```
int pthread_join_internal(pthread_t thread, FAR pthread_addr_t *pexit_value, bool blocking)
{
	... <snip> ...
	if ((pid_t)thread == rtcb->pid) {
		return EDEADLK;
	}
	... <snip> ...
}
```
In this case, checking return value is necessary.